crates/typst-library/src/foundations/func.rs

```diff
@@ -42,47 +40,35 @@ use crate::foundations::{
 /// #list[A][B]
 /// ```
 ///
-/// Functions are a fundamental building block of Typst. Typst provides
-/// functions for a variety of typesetting tasks. Moreover, the markup you write
-/// is backed by functions and all styling happens through functions. This
-/// reference lists all available functions and how you can use them. Please
-/// also refer to the documentation about [set]($styling/#set-rules) and
-/// [show]($styling/#show-rules) rules to learn about additional ways you can
-/// work with functions in Typst.
+/// 関数はTypstにおいて基礎となる構成要素です。
+/// Typstは様々な組版タスクに応じた関数を提供しています。
+/// さらには、作成されるマークアップの裏側では関数が用いられており、すべてのスタイル設定は関数を介して行われます。
+/// このリファレンスでは利用可能なすべての関数とその使い方を示します。
+/// Typstで関数をさらに活用する方法については、[set]($styling/#set-rules)ルールおよび[show]($styling/#show-rules)ルールのドキュメントも参照して下さい。
 ///
-/// # Element functions
-/// Some functions are associated with _elements_ like [headings]($heading) or
-/// [tables]($table). When called, these create an element of their respective
-/// kind. In contrast to normal functions, they can further be used in [set
-/// rules]($styling/#set-rules), [show rules]($styling/#show-rules), and
-/// [selectors]($selector).
+/// # 要素関数
+/// [見出し]($heading)や[表]($table)のような、いくつかの関数は _要素_ と結びついており、呼び出すとその種類に応じた要素を作成します。
+/// さらに、通常の関数とは異なり、要素関数は[setルール]($styling/#set-rules)、[showルール]($styling/#show-rules)および [セレクター]($selector)で使用可能です。
 ///
-/// # Function scopes
-/// Functions can hold related definitions in their own scope, similar to a
-/// [module]($scripting/#modules). Examples of this are
-/// [`assert.eq`]($assert.eq) or [`list.item`]($list.item). However, this
-/// feature is currently only available for built-in functions.
+/// # 関数スコープ
+/// 関数は、[モジュール]($scripting/#modules)と同様に自身のスコープ内に関連する定義を保持できます。
+/// この例は[`assert.eq`]($assert.eq)や[`list.item`]($list.item)です。
+/// しかし、現在この機能が利用可能なのは組み込み関数のみです。
 ///
-/// # Defining functions
-/// You can define your own function with a [let binding]($scripting/#bindings)
-/// that has a parameter list after the binding's name. The parameter list can
-/// contain mandatory positional parameters, named parameters with default
-/// values and [argument sinks]($arguments).
+/// # ユーザー定義関数
+/// [letバインディング]($scripting/#bindings)を用いることで、バインディング名の後に引数リストを持ったユーザー定義関数を定義することができます。
+/// 引数リストには必須の位置引数、デフォルト値を持つキーワード引数および[可変長引数]($arguments)を用いることができます。
 ///
-/// The right-hand side of a function binding is the function body, which can be
-/// a block or any other expression. It defines the function's return value and
-/// can depend on the parameters. If the function body is a [code
-/// block]($scripting/#blocks), the return value is the result of joining the
-/// values of each expression in the block.
+/// 関数バインディングの右辺は関数本体で、ブロックか任意の式です。
+/// 関数の戻り値を定義し、引数に依存させることができます。
+/// 関数本体が[コードブロック]($scripting/#blocks)の場合、戻り値はブロック内の各式を結合させた結果になります。
```

Comment:
非常に細かいコメントですが、ここでは「複数の式を**上から順番に評価**し、それらの**最終的な結果**を返す」というニュアンスが出たほうがよいかなと思ったため、「各式」よりも「すべての式」、「結合」よりも「連結」という言葉を使った方がイメージがわくと思いました。

```suggestion
/// 関数本体が[コードブロック]($scripting/#blocks)の場合、戻り値はブロック内のすべての式を連結させた結果になります。
```